fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -773,8 +773,10 @@ private void handleChargePaidTransaction(final LoanCharge charge, final LoanTran
         LocalDate startDate = getDisbursementDate();
         List<LoanRepaymentScheduleInstallment> installments = getRepaymentScheduleInstallments();
         for (final LoanRepaymentScheduleInstallment installment : installments) {
-            if (installmentNumber == null && charge.isDueForCollectionFromAndUpToAndIncluding(startDate, installment.getDueDate())) {
-
+            boolean isDue = installment.isFirstPeriod()
+                    ? charge.isDueForCollectionFromIncludingAndUpToAndIncluding(startDate, installment.getDueDate())
```

Comment:
Can both the cases be handled using `installment.getFromDate(), installment.getDueDate()`?

Reply:
Probably yes.